### PR TITLE
Add responses generator to test-common lib.

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Integration tests should have dependency only
from public dataservice api. Move generating
urls and responses to test-common lib.

Relates-To: OLPEDGE-1901

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: tests/common/UrlGenerators.h [hunk 1]
```diff
@@ -22,50 +22,26 @@
...
-    uint64_t version) {
-  std::string path = "/layers/" + layer + "/partitions?";
-  for (const auto& partition : partitions) {
-    path.append("partition=" + partition + "&");
-  }
-  path.append("version=" + std::to_string(version));
-  return path;
-}
+class PathGenerator {
+ public:
```
Comment (kzvieriev) [inline tests/common/UrlGenerators.h:29]: Class and file naming is confusing. Lets name both `UrlGenerator` or `PathGenerator`.

File: tests/common/GenerateStringResponses.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  str[str.length() - 1] = ']';
+  return str;
+}
+}  // namespace
+
+std::string ResponseGenerator::ResourceApis(const std::string& catalog) {
+  auto api_response =
+      mockserver::ApiDefaultResponses::GenerateResourceApisResponse(catalog);
+  return serialize(api_response);
+}
```
Comment (kzvieriev) [inline tests/common/GenerateStringResponses.cpp:55]: Why we need custom serialize here?
We have serialize for vector and for api, so should be no problem to use`olp::serializer::serialize(api_response)`

File: tests/common/UrlGenerators.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+std::string PathGenerator::GetQuadKey(const std::string& quadkey,
+                                      const std::string& layer,
+                                      uint64_t version, uint64_t depth) {
+  return "/layers/" + layer + "/versions/" + std::to_string(version) +
+         "/quadkeys/" + quadkey + "/depths/" + std::to_string(depth);
+}
+
+std::string PathGenerator::FullPath(const std::string& apis_response,
```
Comment (kzvieriev) [inline tests/common/UrlGenerators.cpp:61]: maybe we can use apis model instead, so no need for converting apis to json and back.

File: tests/common/GenerateStringResponses.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <olp/dataservice/read/model/Partitions.h>
+
+class ResponseGenerator {
```
Comment (kzvieriev) [inline tests/common/GenerateStringResponses.h:27]: Same here. I think `ResponseGenerator.h` suits better :-)

File: tests/common/UrlGenerators.cpp (hunk 3 above)
Comment (LiubovDidkivska) [reply tests/common/UrlGenerators.cpp:61]: In this case we will have dependancy in integration tests fron src. Here we want to keep all such dependancies in test-common lib.

**Review by kzvieriev — DISMISSED** [review]

File: tests/common/ResponseGenerator.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#include "PlatformUrlsGenerator.h"
+#include "ReadDefaultResponses.h"
+#include "generated/model/Api.h"
+// clang-format off
+#include "generated/serializer/ApiSerializer.h"
+#include "generated/serializer/VersionResponseSerializer.h"
+#include "generated/serializer/PartitionsSerializer.h"
+#include <olp/core/generated/serializer/SerializerWrapper.h>
+#include "generated/serializer/JsonSerializer.h"
+
```
Comment (kzvieriev) [inline tests/common/ResponseGenerator.cpp:37]: // clang-format on

**Review by meravingen123 — APPROVED** [review]

**Review by kzvieriev — APPROVED** [review]

**Pull request merged**